projects/hathor-wallet-headless/hsm-integration.md

```diff
@@ -0,0 +1,280 @@
+hsmPassword: "my_password"
+```
+
+One note about this is that a connection may be open at the time the end user decides to reload the configuration settings. For now, this will not be allowed as it would increase the complexity of dealing with closing the current connections. Other connection setting approaches are discussed in the "Future Possibilities" section below.
+
+A dedicated `[POST] /hsm/start` endpoint will be available, receiving the expected `wallet-id` and `hsm-key` body parameters. The `hsm-key` indicates which key on the HSM contains the BIP32 information for this wallet. The headless considers this to be a _hardware wallet_ and realizes the first request to the HSM to validate the existence of the key name.
+
+> **Possibility 1: The key is available**<br>
+> The case flow continues
+>
+> **Possibility 2: The key is not available**<br>
+> An error is thrown indicating the HSM has not been properly configured to have a BIP32 `xPriv` on this specific key.
+
+The key being available, the next call requests the wallet's `xPub`. The parsing of `xPub` information received from the HSM can be based on the desktop hardware wallet [`handlePublicKeyData` function](https://github.com/HathorNetwork/hathor-wallet/blob/ef57015015375a477cffd72baf62f4e14baf541a/src/screens/StartHardwareWallet.js#L106-L138).
+
+If the generated `xPub` is not valid, an error is thrown to the user indicating that the designated HSM key does not contain a valid `BIP32 xPriv` and is unsuitable for using as a wallet.
+
+Once the wallet is started successfully, all other operations are executed in a similar fashion to the current headless implementation of a read-only wallet.
+
+## Signature integration - Phase 2
+The signature of HSM inputs will be made through a `Tx Proposal`, using the Atomic Swap API. The `wallet/atomic-swap/get-my-signatures` route will be modified to identify that the wallet is related to an HSM and will send the data to be signed to the *HSM Client*. This can be based on the desktop hardware wallet [`ledger/sendTx()` function](https://github.com/HathorNetwork/hathor-wallet/blob/ef57015015375a477cffd72baf62f4e14baf541a/src/utils/ledger.js#L126-L166).
+
+Only this route will be adapted for using the hardware wallet, since it's the only one that has an implementation close enough to the Desktop Ledger. The other signature endpoints, such as `wallet/simple_send_tx` would require more complex refactorings and will be discussed on the _Future Possibilites_ section.
+
+## Creating a BIP32 on the HSM
+This operation should be done via script, as it is the most critical operation on the HSM within this scope.
+
+As such, a new `scripts/createHSMKey.js` file will be created, offering a way to quickly generate a new BIP32 wallet on a developer machine. This created key name should be inserted on the `config.js` file along with a `walletId` for it. Other wallet generation possibilities are discussed on the _Alternatives_ section of this document.
```

Comment:
I didn't understand this. What will be added to the `config.js`? Which field and why?

Reply:
Previous iterations of this design had the keys set on the `config.js`, this is a leftover from these older versions.
Fixed on [12985ed](https://github.com/HathorNetwork/rfcs/pull/67/commits/12985ed8320681ac512326809f776bee792fca88).